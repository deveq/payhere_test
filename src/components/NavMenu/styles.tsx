import { NavLink } from "react-router-dom"
import styled from "styled-components"

export const List = styled.ul`
    display: flex;
    flex-direction: row;
    flex: 1;
`;

export const ListItem = styled.li`
    height: 30px;
    width: 100px;
    display: flex;
    align-items: center;
    justify-content: center;
`
export const Link = styled(NavLink)`
    color: ${({theme}) => theme.text1};
    text-decoration: none;
    font-size: 20px;
    text-transform: capitalize;
    &:hover {
        opacity: 0.6;
    }
    &.active {
        font-weight: bold;
        text-decoration: underline;
    }
`;
