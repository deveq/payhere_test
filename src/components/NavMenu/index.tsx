import * as Styled from './styles';

export interface List {
	to: string;
	title: string;
}

interface NavMenuProps {
	list: List[];
}

const NavMenu = ({ list }: NavMenuProps) => {
	return (
		<Styled.List>
			{list.map((item) => (
				<Styled.ListItem key={item.title}>
					<Styled.Link to={item.to}>{item.title}</Styled.Link>
				</Styled.ListItem>
			))}
		</Styled.List>
	);
};

export default NavMenu;
