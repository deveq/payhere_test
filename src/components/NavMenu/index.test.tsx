import {screen} from '@testing-library/react';
import {renderWithWrapper} from 'tests';
import NavMenu from './index';

describe('<NavMenu />', () => {
    const data = [
        {title: 'title1', to: 'to1'},
        {title: 'title2', to: 'to2'},
        {title: 'title3', to: 'to3'},
    ]
    it('정상적으로 렌더링되었는지 확인한다', () => {
        const {container} = renderWithWrapper(<NavMenu list={data} />);

        const list = screen.getByRole('list');
        expect(list).toBeInTheDocument();
        expect(list.childElementCount).toBe(data.length);
        expect(container).toMatchSnapshot();
    })
});