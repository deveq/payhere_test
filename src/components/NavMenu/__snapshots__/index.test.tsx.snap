// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavMenu /> 정상적으로 렌더링되었는지 확인한다 1`] = `
<div>
  <ul
    class="sc-bcXHqe jZUBXE"
  >
    <li
      class="sc-gswNZR bnXthA"
    >
      <a
        class="sc-dkrFOg byeDqj"
        href="/to1"
      >
        title1
      </a>
    </li>
    <li
      class="sc-gswNZR bnXthA"
    >
      <a
        class="sc-dkrFOg byeDqj"
        href="/to2"
      >
        title2
      </a>
    </li>
    <li
      class="sc-gswNZR bnXthA"
    >
      <a
        class="sc-dkrFOg byeDqj"
        href="/to3"
      >
        title3
      </a>
    </li>
  </ul>
</div>
`;
